dll/win32/shell32/wine/control.c

```diff
@@ -736,6 +737,75 @@ static	void	Control_DoWindow(CPanel* panel, HWND hWnd, HINSTANCE hInst)
 }
 #endif
 
+#ifdef __REACTOS__
+
+/** Structure for in and out data when
+ *  search for the cpl dialog of first instance
+ */
+typedef struct tagAppDlgFindData
+{
+    PCWSTR  szAppFile;  /**< Full path to applet library as search parameter */
+    UINT    sAppletNo;  /**< Number of applet in a system control library as search parameter */
+    ATOM    aCPLName;   /**< to read window property 'CPLName' */
+    ATOM    aCPLFlags;  /**< to read window property 'CPLFlags'*/
+    HWND    hRunDLL;    /**< to skip self instance */
+    HWND    hDlgResult; /**< Returned dialog handle or NULL if not found */
+} AppDlgFindData;
+
+/**
+ * Callback function to search applet dialog
+ * @param hwnd A handle to a top-level window.
+ * @param lParam Pointer of AppDlgFindData
+ * @return TRUE: continue enumeration, FALSE: stop enumeration
+ */
+static BOOL CALLBACK
+Control_EnumWinProc(
+    _In_ HWND   hwnd,
+    _In_ LPARAM lParam)
+{
+    AppDlgFindData* pData = (AppDlgFindData*)lParam;
+    WCHAR szClassName[256] = L"";
+
+    if (pData->hRunDLL == hwnd)
+    {
+        // Skip self instance
+        return TRUE;
+    }
+
+    if (GetClassNameW(hwnd, szClassName, _countof(szClassName)))
+    {
+        // Note: A comparison on identical is not possible, the class names are different.
+        // ReactOS: 'rundll32_window'
+        // WinXP: 'RunDLL'
+        // other OS: not checked
+        if (StrStrIW(szClassName, L"rundll32") != NULL)
+        {
+            UINT sAppletNo;
+
+            sAppletNo = (UINT)GetPropW(hwnd, (LPTSTR)MAKEINTATOM(pData->aCPLFlags));
```

Comment:
We could fix the x64 build error if we used an UINT_PTR for sAppletNo (here + in the structure definition above).

Reply:
That's precisely what will make the build fail. Because you are going to truncate a "HANDLE" (which has the size of a platform-sized integer, i.e. 32 or 64 bits), to a fixed size.